### ISSUE TITLE
Stop using stack

### ISSUE THREAD

**Issue by ndmitchell** [lunaryorn/vscode-hlint#9]
The code currently does `stack exec hlint ...`. Why? Why not just require the user has installed `hlint` and then just run `hlint` directly? It would also mean they could install the latest HLint, and use the up-to-date hlint with code targetting an older stack (or in my case) no stack at all.

**Comment by lunaryorn** [comment]
It used to fit my workflow better.  I wrote this for myself, and I just found it easier to have hlint in local stack environments.   I like this idea that everything's just a dependency of the build, and that the build doesn't depend on any outside tools.  Makes things a lot more reliable in my experience, and it's the way things are in Scala and TypeScript, my main languages these days.

But nowadays I don't use Haskell anymore, so I'm fine w/ changing this code.  I won't do it myself tho, but if you want I can give you commit access to the repo so you can change it yourself.  I think I can even give you rights to push new releases to the extension gallery so you won't need me involved at all to make this extension as you like.

**Comment by ndmitchell** [comment]
Given you're not using it much nowadays, might it be better for me to fork it, maintain it myself, and upload it as a separate extension? If you're not maintaining it my inclination is to go a lot more violent than just changing from stack. (Of course, I would be happy to credit you as a co-author indefinitely, given that it's mostly your work, and will likely remain so forever more.)

**Comment by ndmitchell** [comment]
And as a license matter, if I did fork, would you be happy for me to relicense it as BSD 2 clause, to fit with the HLint license?

**Comment by lunaryorn** [comment]
@ndmitchell Feel free to make your violent changes right here :wink:

As far as I'm concerned you might also fork, but that'd but the third hlint extension into the extension gallery.  Kinda confusing for users isn't it?

As far as the license is concerned, feel free to use whatever you like (just check w/ dependencies, I've got no idea what libraries I've been using here).

**Comment by lunaryorn** [comment]
Ah, that said, I just realized you can apparently unpublish extensions, so if you'd fork I'd try and get this extension removed from the gallery, or at least refer it to yours.

**Comment by ndmitchell** [comment]
My inclination is to move it inside the HLint repo itself, much like I've done with ghcid, which seems to be a reasonable model.

**Comment by lunaryorn** [comment]
Good idea!  Tell me when you've pushed the first version and I'll pull my extension from the gallery and link the README to yours :blush:

**Comment by letmaik** [comment]
Is it really a good idea to move IDE extensions into the repository of the tool? I find that this makes issue tracking more tricky and also demotes extensions a bit, as they don't have a prominent space anymore but just live in some subfolder. Not to mention that sometimes I want to give kudos (=stars) to an extension and not the tool itself, or the other way around.

**Comment by ndmitchell** [comment]
As I think further, I might integrate hlint into ghcid, and then have the ghcid extension just pick up both. At the moment I have the curious case where I get hlint errors on typing but type errors on save - so I see hlint errors that shouldn't be fixed first. If so, I'll still recommend people at this separate extension as is (or another, if anyone else wants to fork it).

**Comment by letmaik** [comment]
By the way, about the original topic of this issue: I'm strongly in favour of keeping the use of stack at least as an option. I find it very useful if you already have a stack workflow.
